examples/ACE/utils/NNIAP.jl

```diff
@@ -78,6 +163,52 @@ function learn!(nniap, ds, opt::Flux.Optimise.AbstractOptimiser, epochs, loss, w
     end
 end
 
+function learn!(nniap, ds, opt::Flux.Optimise.AbstractOptimiser, epochs, loss, w_e, w_f,_device)
+    if _device != gpu
+        return learn!(nniap, ds, opt, epochs, loss, w_e, w_f)
+    end
+    optim = Flux.setup(opt, nniap.nn)
+    nniap.nn = nniap.nn|> gpu
+    nniap.iap = nniap.iap|> gpu
+    ds  = ds |> gpu
+    optim = Flux.gpu(optim)
+
+    ∇gpu_loss(nn, iap, ds, w_e, w_f) = gradient((nn) -> gpu_loss(nn, iap, ds, w_e, w_f), nn)
+    losses = []
+    for epoch in 1:epochs
+        # Compute gradient with current parameters and update model
+        grads = ∇gpu_loss(nniap.nn, nniap.iap, ds, w_e, w_f)
+        Flux.update!(optim, nniap.nn, grads[1])
+        # Logging
+        curr_loss = gpu_loss(nniap.nn, nniap.iap, ds, w_e, w_f)
+        push!(losses, curr_loss)
+        println(curr_loss)
+    end
+end
+
+function learn!(nn, iap, ds, opt::Flux.Optimise.AbstractOptimiser, epochs, loss, w_e, w_f, true_energy, local_descriptors, _device=gpu, n_batches=1) # new
```

Comment:
Use this interface like this:

`learn!(nace, ds, opt::Flux.Optimise.AbstractOptimiser, n_epochs, n_batches = 1, loss, w_e, w_f, device = gpu)`